examples/DataApi-NFT/template/modules/shared_data/src/main/scala/com/my/nft_example/shared_data/Utils.scala

```diff
@@ -1,19 +1,43 @@
 package com.my.nft_example.shared_data
 
+import cats.data.NonEmptySet
+import cats.effect.Async
+import cats.syntax.foldable.toFoldableOps
+import cats.syntax.traverse.toTraverseOps
+import org.tessellation.schema.address.Address
+import org.tessellation.security.SecurityProvider
+import org.tessellation.security.signature.signature.SignatureProof
+
 import java.net.{MalformedURLException, URISyntaxException, URL}
+import scala.util.{Failure, Success, Try}
 
 object Utils {
 
   @throws[MalformedURLException]
   @throws[URISyntaxException]
-  def isValidURL(url: String): Boolean = try {
-    new URL(url).toURI
-    true
-  } catch {
-    case _: MalformedURLException =>
-      false
-    case _: URISyntaxException =>
-      false
+  def isValidURL(url: String): Boolean =
+    Try(new URL(url).toURI) match {
+      case Failure(_) => false
+      case Success(_) => true
+    }
+
+  def getAllAddressesFromProofs[F[_] : Async : SecurityProvider](
+    proofs: NonEmptySet[SignatureProof]
+  ): F[List[Address]] = {
+    proofs
+      .map(_.id)
+      .toList
+      .traverse(_.toAddress[F])
+  }
+
+  def getFirstAddressFromProofs[F[_] : Async : SecurityProvider](
+    proofs: NonEmptySet[SignatureProof]
+  ): F[Address] = {
+    proofs
+      .map(_.id)
+      .toList
+      .head
+      .toAddress[F]
```

Comment:
```suggestion
    proofs.head.id.toAddress[F]
```
No need to construct a new Set and then a List if we only want the first element anyway

Reply:
Got it